### PR TITLE
 Fix a bug in usage of windows-style paths as SHELL path.

### LINKED ISSUE #16646: specifying shell/$SHELL path on windows
### Neovim version (nvim -v)

0.6.0

### Vim (not Nvim) behaves the same?

no 8.2.3441

### Operating system/version

Windows 10/19043.1348

### Terminal name/version

Windows terminal 1.11.2921.0

### $TERM environment variable

none in powershell, xterm-256color in git-bash

### Installation

scoop

### How to reproduce the issue

1 Run `powershell` or `mingw-bash.exe` in windows terminal or the stand-alone `neovim` app.
2 Try and set `shell` to a path.

### Expected behavior

According to the [shell documentation](https://neovim.io/doc/user/options.html#'shell'), one can specify the shell pathname like this:

```
Note the backslash before each quote (to avoid starting a comment) and 
	each space (to avoid ending the option value), so better use |:let-&| 
	like this:
		:let &shell='"C:\Program Files\unix\sh.exe" -f'
```
It should be noted that I'm trying to set `C:\Program Files\Git\bin\bash.exe -i -l` as shell but since it has spaces I'd first expect to be able to set the more basic `C:\Windows\System32\cmd.exe` first.

Ideally, this would work just like in `vim` where my `SHELL` is `/usr/bin/bash` and everything works.
If that's not possible it'd be great if there's a way to manually set my shell to `C:\Program Files\Git\usr\bin\bash`.

### Actual behavior

# Setting the environment SHELL variable to a path
In `powershell`
```ps
$Env:SHELL = "C:\Windows\System32\cmd.exe"
```
In `git-bash`
```sh
env SHELL='C:\Windows\System32\cmd.exe' nvim
```
## result
```vim
:echo $SHELL
C:\Windows\System32\cmd.exe
:ter
E475: Invalid value for argument cmd: 'C:WindowsSystem32cmd.exe' is not executable
```
# Using let-& (wo environment variable)
```vim
:let &shell='"C:\Windows\System32\cmd.exe"'
:ter
E475: Invalid value for argument cmd: 'C:WindowsSystem32cmd.exe' is not executable
```
# Using let-& forward slashes
```vim
:let &shell='"c:/Windows/System32/cmd.exe"'
A subdirectory or file .exe already exists.
[Process exited 1]
term://~//28128:c:/Windows/System32/cmd.exe
```
# Using a path with spaces
```vim
:let &shell='"C:\Program Files\Git\bin\bash"'
E475: Invalid value for argument cmd: 'C:Program FilesGitusr^Hin^Hash.exe' is not executable
```
# Specifying bash in nvim in `git-bash`
```
:ter bash
/usr/bin/bash: /s: No such file or directory
[Process exited 127]
```

### LINKED ISSUE #25776: :terminal generates E475 if $SHELL is set to an absolute path on Windows
### Problem

`:terminal` generates error E475 if the environment variable `SHELL` is set to an absolute path on Windows. For example, if `SHELL` is `C:\msys64\usr\bin\fish.exe`, the editor throws `E475: Invalid value for argument cmd: 'C:msys64usr^Hin^Lish.exe' is not executable`

### Steps to reproduce

1. set the environment variable `SHELL` to the absolute path of a valid shell.
2. run Neovim
3. enter `:terminal`

### Expected behavior

A terminal should open in the editor.

### Neovim version (nvim -v)

v0.9.4

### Vim (not Nvim) behaves the same?

N/A

### Operating system/version

Windows 11

### Terminal name/version

Windows Terminal 1.18.2822.0

### $TERM environment variable

xterm-256color

### Installation

Windows Installer

### PR DESCRIPTION
**Pull request by luki446**
Fix #16646
Fix #25776

### CODE REVIEW

File: src/nvim/option.c [hunk 1]
```diff
@@ -180,14 +180,37 @@ static void set_init_default_shell(void)
...
   // Don't use it if it is empty.
   const char *shell = os_getenv("SHELL");
+
   if (shell != NULL) {
+  char * fixed_shell = NULL;
+
+  // In case of using backslashes in SHELL path switch them to forward slashes
+  // so given shell can be run.
+#ifdef BACKSLASH_IN_FILENAME
+  if(vim_strchr(shell, '\\') != NULL) {
```
Comment (justinmk) [inline src/nvim/option.c:190]: is this the inverse of slash_adjust? https://github.com/neovim/neovim/blob/5d4d5b5d54dc2e46733d25bfa2ee80a35db76b39/src/nvim/path.c#L1423

inlining a random special case in `set_init_default_shell` doesn't seem very maintainable.

also needs a test.

Comment (luki446) [reply src/nvim/option.c:190]: Thank you for your comment, sooo how would you implement it differently?

Comment (zeertzjq) [reply src/nvim/option.c:190]: Does changing `ex_terminal()` (e.g. <https://github.com/neovim/neovim/issues/16646#issuecomment-993599308>) not work?

**Review by dundargoc — APPROVED** [review]
Works for me locally.

**Comment by luki446** [comment]
Also fix #25776

**Pull request merged**